packages/tools/src/tools/segmentation/BrushTool.ts

```diff
@@ -122,30 +125,88 @@ class BrushTool extends BaseTool {
       );
     }
 
-    const { segmentationId, type } = activeSegmentationRepresentation;
+    const { segmentationId, type, segmentationRepresentationUID } =
+      activeSegmentationRepresentation;
+
+    if (type === SegmentationRepresentations.Contour) {
+      throw new Error('Not implemented yet');
+    }
+
     const segmentsLocked = segmentLocking.getLockedSegments(segmentationId);
 
     const { representationData } =
       segmentationState.getSegmentation(segmentationId);
 
-    // Todo: are we going to support contour editing with this tool?
-    const { volumeId } = representationData[type] as LabelmapSegmentationData;
-    const segmentation = cache.getVolume(volumeId);
+    const labelmapData =
+      representationData[SegmentationRepresentations.Labelmap];
 
-    const actors = viewport.getActors();
+    const viewportIdsToRender = [viewport.id];
 
-    // Note: For tools that need the source data. Assumed to use
-    // First volume actor for now.
-    const firstVolumeActorUID = actors[0].uid;
-    const imageVolume = cache.getVolume(firstVolumeActorUID);
+    if (isVolumeSegmentation(labelmapData)) {
+      const { volumeId } = representationData[
+        type
+      ] as LabelmapSegmentationDataVolume;
+      const actors = viewport.getActors();
+
+      // Note: For tools that need the source data. Assumed to use
+      // First volume actor for now.
+      const firstVolumeActorUID = actors[0].uid;
+
+      this._editData = {
+        volumeId,
+        referencedVolumeId: firstVolumeActorUID,
+        segmentsLocked,
+        segmentationRepresentationUID,
+      };
+    } else {
+      const { imageIds: segmentationImageIds } =
+        labelmapData as LabelmapSegmentationDataStack;
+
+      // const { zSpacing, sortedImageIds, origin } =
+      //   sortImageIds(referencedImageIds);
+
+      // const orderedSegmentationImageIds = sortedImageIds.map((imageId) =>
+      //   getDerivedImageId(imageId, referencedImageIds, segmentationImageIds)
+      // );
+
+      const currentImageId = viewport.getCurrentImageId();
+      const currentSegmentationImageId =
+        segmentationImageIds.indexOf(currentImageId);
+
+      if (!currentSegmentationImageId) {
+        throw new Error('No current segmentation image id');
+      }
 
-    const viewportIdsToRender = [viewport.id];
+      // here we should identify if we can perform sphere manipulation
+      // for these stack of images, if the metadata is not present
+      // to create a volume or if there are inconsistencies between
+      // the image metadata we should not allow the sphere manipulation
+      // and should throw an error or maybe simply just allow circle manipulation
+      // and not sphere manipulation
```

Comment:
IMO it should not allow the sphere manipulation and throw an error because the tool shouldn't be even enabled on the toolbar when the active viewport is a stack viewport